wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -275,6 +276,20 @@ public void add(AddParams params) throws GitException {
 
         try {
             addCommand.call();
+
+            // to add deleted files in index it is required to perform git rm on them
```

Comment:
If you need a comment it means that code excerpt barely can be understood. Introduce a new method.